mex/backend/identity/main.py

```diff
@@ -0,0 +1,54 @@
+from fastapi import APIRouter
+from fastapi.responses import JSONResponse
+
+from mex.backend.graph.connector import GraphConnector
+from mex.backend.graph.transform import transform_identity_result_to_identity
+from mex.backend.identity.models import IdentityAssignRequest, IdentityFetchResponse
+from mex.backend.transform import to_primitive
+from mex.common.exceptions import MExError
+from mex.common.identity.models import Identity
+from mex.common.types import Identifier
+
+router = APIRouter()
+
+
+@router.post("/identity", status_code=200, tags=["extractors"])
+def assign_identity(request: IdentityAssignRequest) -> Identity:
+    """Insert a new identity or update an existing one."""
+    connector = GraphConnector.get()
+    graph_result = connector.fetch_identities(
+        had_primary_source=request.hadPrimarySource,
+        identifier_in_primary_source=request.identifierInPrimarySource,
+    )
+    if len(graph_result.data) > 1:
+        raise MExError("found multiple identities indicating graph inconsistency")
+    if len(graph_result.data) == 1:
+        identity = transform_identity_result_to_identity(graph_result.data[0])
+    else:
+        identity = Identity(
+            hadPrimarySource=request.hadPrimarySource,
+            identifier=Identifier.generate(),
+            identifierInPrimarySource=request.identifierInPrimarySource,
+            stableTargetId=Identifier.generate(),
+        )
+    return JSONResponse(to_primitive(identity), 200)  # type: ignore[return-value]
```

Comment:
casting would not be quite correct though, because we are not returning an object of class `Identity`.

i'm also not happy with this, but i get errors from fastapi like this when i return the `identity` instance directly:

`Error 1 validation errors:
  {'loc': ('response', '__root__'), 'msg': 'got multiple values for items', 'type': 'value_error'}`

not sure exactly what is happening here, but i hope it'll be fixed with pydantic 2

Reply:
I see, can we return identity directly instead of JSON response? because 200 is already defined in the decorator and if we return `identity` object it'll be converted to JSON by fast api anyway.

Reply:
like i said, i tried that. but it throws `Error 1 validation errors: {'loc': ('response', '__root__'), 'msg': 'got multiple values for items', 'type': 'value_error'}`, which i was not able to debug. but i'm hoping it'll go away with pydantic 2